packages/ra-core/src/dataProvider/cacheDataProviderProxy.ts

```diff
@@ -23,16 +23,30 @@ export default (
     duration: number = fiveMinutes
 ): DataProvider =>
     new Proxy(dataProvider, {
-        get: (target, name: string) => (resource, params) => {
-            if (name === 'getList' || name === 'getMany' || name === 'getOne') {
-                // @ts-ignore
-                return dataProvider[name](resource, params).then(response => {
-                    const validUntil = new Date();
-                    validUntil.setTime(validUntil.getTime() + duration);
-                    response.validUntil = validUntil;
-                    return response;
-                });
+        get: (target, name: string) => {
+            if (typeof name === 'symbol') {
+                return;
+            } else {
```

Comment:
Could you remove the `else` in all your modifications? It's useless because of the early return, and it harms legibility as it adds one level of indentation. Besides, it would make your changes smaller and easier to review!

Reply:
Done.